projects/rocprofiler-sdk/source/lib/python/rocpd/csv.py

```diff
@@ -23,14 +23,401 @@
+            scratch_size AS Scratch_Size,
+            vgpr_count AS VGPR_Count,
+            accum_vgpr_count AS Accum_VGPR_Count,
+            sgpr_count AS SGPR_Count,
+            counter_name AS Counter_Name,
+            value AS Counter_Value,
+            start AS Start_Timestamp,
+            end AS End_Timestamp
+        FROM "counters_collection"
+        ORDER BY
+            guid ASC, start ASC, end DESC
+    """
+    write_sql_query_to_csv(
+        importData, query, config.output_path, config.output_file, "counter_collection"
+    )
+
+
+def write_scratch_memory_csv(importData, config) -> None:
+
+    if config.agent_index_value == libpyrocpd.agent_indexing.node:  # absolute
+        agent_id = "'Agent ' || agent_abs_index"
+    elif (
+        config.agent_index_value == libpyrocpd.agent_indexing.logical_node
+    ):  # relative (default)
+        agent_id = "'Agent ' || agent_log_index"
+    elif (
+        config.agent_index_value == libpyrocpd.agent_indexing.logical_node_type
+    ):  # type-relative
+        agent_id = "agent_type || ' ' || agent_type_index"
+    else:
+        agent_id = ""
+
+    query = f"""
+        SELECT
+            guid AS Guid,
+            'SCRATCH_MEMORY' AS Kind,
+            'SCRATCH_MEMORY_' || operation AS Operation,
+            {agent_id} AS Agent_Id,
+            queue_id AS Queue_Id,
+            tid AS Thread_Id,
+            alloc_flags AS Alloc_Flags,
+            start AS Start_Timestamp,
+            end AS End_Timestamp
+        FROM "scratch_memory"
+        ORDER BY
+            guid ASC, start ASC, end DESC
+    """
+    write_sql_query_to_csv(
+        importData, query, config.output_path, config.output_file, "scratch_memory"
+    )
+
+
+def write_region_csv(importData, config) -> None:
+
+    query = """
+        SELECT
+            guid AS Guid,
+            category AS Domain,
+            name AS Function,
+            pid AS Process_Id,
+            tid AS Thread_Id,
+            stack_id AS Correlation_Id,
+            start AS Start_Timestamp,
+            end AS End_Timestamp
+        FROM "regions"
+        WHERE
+            category LIKE 'HIP_%' OR
+            category LIKE 'HSA_%' OR
+            category LIKE 'RCCL_%' OR
+            category LIKE 'ROCDECODE_%' OR
+            category LIKE 'ROCJPEG_%'
+        UNION ALL
+                SELECT
+            guid AS Guid,
+            category AS Domain,
+            CASE
+                WHEN json_extract(extdata, '$.message') IS NOT NULL
+                THEN json_extract(extdata, '$.message')
+                ELSE name
+            END AS Function,
+            pid AS Process_Id,
+            tid AS Thread_Id,
+            stack_id AS Correlation_Id,
+            start AS Start_Timestamp,
+            end AS End_Timestamp
+        FROM "regions_and_samples"
+        WHERE
+            category LIKE 'MARKER_%'
+        ORDER BY
+            guid ASC, start ASC, end DESC
```

Comment:
```suggestion
    query = """
        SELECT
            guid AS Guid,
            category AS Domain,
            name AS Function,
            pid AS Process_Id,
            tid AS Thread_Id,
            stack_id AS Correlation_Id,
            start AS Start_Timestamp,
            end AS End_Timestamp
        FROM "regions_and_samples"
        ORDER BY
            guid ASC, start ASC, end DESC
```
Even simpler than your change.  You can just dump out the `region `view, or `region_and_samples`.  I found regions was not enough when testing against the JSON, so `region_and_samples` included what is listed in JSON, I think.  Part of Jonathan's reasoning of getting rid of _API CSVs is that the string comparison `category LIKE 'RCCL_%` runs quite slowly in SQLite.  He's suggesting we avoid the slow queries.  

Instead, if there's just a regions CSV, then the user can filter what they need themselves in Excel, etc.